### PR TITLE
Fix Migration Errors

### PR DESCRIPTION
**Pull request by justinmc**
This PR fixes errors that would occur during db initialization for a fresh database.  It also fixes an error logging bug in the process!

### CODE REVIEW

File: server/bin/www [hunk 1]
```diff
@@ -5,7 +5,7 @@
  */
 if (!('toJSON' in Error.prototype)) {
   Object.defineProperty(Error.prototype, 'toJSON', {
-    value: () => {
+    value: function toJSON() {
```
Comment (justinmc) [inline server/bin/www]: The arrow function was injecting the parent context into itself, so `this` below didn't reference the error.  Using a regular function fixed the bug and allowed toJSON to give back meaningful stuff.

File: server/utils/db_utils.js [hunk 2]
```diff
@@ -64,7 +66,27 @@ const dbUtils = {
...
+      });
+    });
+  },
+
+  /**
+   * Set the db version to the latest value
+   * @param {RedisClient} redis
+   * @returns {Promise}
+   */
+  seedVersion(redis) {
```
Comment (justinmc) [inline server/utils/db_utils.js]: I wasn't explicitly seeding the db version number in some cases, so this fixes that.

File: server/utils/migrations_utils.js [hunk 3]
```diff
@@ -7,52 +8,58 @@ const migrationsUtils = {
    * Rename an old field in each run to the new name
    */
   migrateRunsWorkflowField(redis) {
-    global.log.debug('Migrating runs workflow fields...');
```
Comment (justinmc) [inline server/utils/migrations_utils.js]: This was the first bug in the migrations.  We moved from a global variable to a `log` module but I guess I forgot to change out these ones.

File: server/utils/migrations_utils.js [hunk 4]
```diff
@@ -7,52 +8,58 @@ const migrationsUtils = {
...
 
           return redis.hset(dbConstants.REDIS_APPS, oldApp.id, JSON.stringify(oldApp)).then(() => {
-            global.log.debug(`Finished migrating workflow ${oldApp.id} to app.`);
-          }).catch(global.log.error);
-        }).catch(global.log.error);
+            log.debug(`Finished migrating workflow ${oldApp.id} to app.`);
+          });
+        });
       });
-    }).catch(global.log.error);
```
Comment (justinmc) [inline server/utils/migrations_utils.js]: I removed all these intermediate error handling things like we discussed and instead handle everything at the root level in bin/www where `dbUtils.inititalize` is called.

File: server/utils/seed_data.js [hunk 5]
```diff
@@ -1,101 +1,105 @@
...
-    description: 'Calculate the electron binding energy of an' +
-        ' anionic doublet species using DFT',
-    widgets: [
-      {
-        id: widgetsConstants.LOAD,
-        meta: {
-          title: 'Load Molecule',
+const seedData = {
+  version: 1,
+  apps: [
```
Comment (justinmc) [inline server/utils/seed_data.js]: I changed `seedData` to an object so I can include seed data other than just apps (in this case I needed the version number).

File: server/utils/migrations_utils.js [hunk 6]
```diff
@@ -7,52 +8,58 @@ const migrationsUtils = {
...
    * Move workflows from the old workflow table into the app table
    */
   migrateWorkflowsToApps(redis) {
-    global.log.debug('Migrating workflows to apps...');
+    log.debug('Migrating workflows to apps...');
 
     return redis.hgetall(OLD_REDIS_APPS).then((oldAppsHash) => {
+      if (!oldAppsHash) {
+        log.debug('Not migrating workflows to apps because no workflows found');
+        return Promise.resolve();
```
Comment (justinmc) [inline server/utils/migrations_utils.js]: The other main bug in migrations was that the migrations would choke if there were no old `workflows` to migrate, such as in a fresh build!

**Comment by justinmc** [comment]
@dionjwa Take a look at my comments above about the main points.  This appears to fix seeding/migrating a fresh db for me, and it also handles errors in that place much better.

File: server/bin/www (hunk 1 above)
Comment (dionjwa) [inline server/bin/www]: Yeah, I've hit this issue before. I didn't realize that arrow functions work differently to named functions .

File: server/bin/www [hunk 7]
```diff
@@ -110,7 +110,7 @@ dbUtils.initialize(redis).then(() => {
   server.on('listening', onListening(server.address()));
   server.timeout = 8 * 60 * 1000; // 8 minutes
 }).catch((err) => {
-  log.error({ error: err });
+  log.error({ error: err.toJSON() });
 });
```
Comment (dionjwa) [inline server/bin/www]: Shouldn't this catch be removed completely? I think the server should crash if there is an unresolved error at this state in the app.

Comment (justinmc) [inline server/bin/www]: Whoops I forgot about that.  I'll add the promise exception event handler and then remove this.

**Pull request merged**